Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserExpandedTableView.cpp

```diff
@@ -439,77 +439,29 @@ namespace AzToolsFramework
             const QVariant text = index.data(Qt::DisplayRole);
             if (text.isValid())
             {
-                QString iconPathToUse;
                 QStyleOptionViewItem options{ option };
                 initStyleOption(&options, index);
+                int height = options.rect.height();
+                QRect iconRect(0, options.rect.y() + 5, height - 10, height - 10);
+                QSize iconSize = iconRect.size();
+                QStyle* style = options.widget ? options.widget->style() : qApp->style();
 
-                const QVariant path = index.data(Qt::UserRole);
-
-                AZ::EBusAggregateResults<SourceFileDetails> results;
-                AssetBrowserInteractionNotificationBus::BroadcastResult(
-                                    results, &AssetBrowserInteractionNotificationBus::Events::GetSourceFileDetails, path.toString().toUtf8().constData());
-
-                auto it = AZStd::find_if( results.values.begin(), results.values.end(),
-                    [](const SourceFileDetails& details)
-                    {
-                        return !details.m_sourceThumbnailPath.empty();
-                    });
-
-                const bool isFolder = index.data(Qt::UserRole + 1).toBool();
-
-                if (it != results.values.end() || isFolder)
+                const auto& qVariant = index.data(Qt::UserRole + 1);
+                if (!qVariant.isNull())
                 {
-                    static constexpr const char* FolderIconPath = "Icons/AssetBrowser/Folder_16.svg";
-                    const char* resultPath = isFolder ? FolderIconPath : it->m_sourceThumbnailPath.c_str();
-                    // its an ordered bus, though, so first one wins.
-                    // we have to massage this though.  there are three valid possibilities
-                    // 1. its a relative path to source, in which case we have to find the full path
-                    // 2. its an absolute path, in which case we use it as-is
-                    // 3. its an embedded resource, in which case we use it as is.
-
-                    // is it an embedded resource or absolute path?
-                    if ((resultPath[0] == ':') || (!AzFramework::StringFunc::Path::IsRelative(resultPath)))
+                    QIcon icon;
+                    if (const auto& path = qVariant.value<QString>(); !path.isEmpty())
                     {
-                        iconPathToUse = QString::fromUtf8(resultPath);
+                        icon.addFile(path, iconSize, QIcon::Normal, QIcon::Off);
+                        AZ_Assert(!icon.isNull(), "Asset Browser Icon not found for file '%s'", path.constData());
```

Comment:
Again the Assert here needs to convert the QString to a utf8 QByteArray, where then `constData` can be used to get a `const char*`
```suggestion
                        AZ_Assert(!icon.isNull(), "Asset Browser Icon not found for file '%s'", path.toUtf8().constData());
```

Reply:
Thanks, fixed!